policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +69,76 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected. When a TCP probe is
+/// configured there is no path, so we use '/'.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    if let Some(spec) = spec {
+        for container in spec.containers.clone().into_iter() {
```

Comment:
do we need to clone the containers here? it seems like we could do all of this by iterating over the containers by reference...